proto/plan_common.proto

```diff
@@ -41,6 +41,9 @@ message StorageTableDesc {
   uint32 retention_seconds = 5;
   repeated uint32 value_indices = 6;
   uint32 read_prefix_len_hint = 7;
+  // Whether the table is versioned. If `true`, column-aware row encoding will be used
+  // to be compatible with schema changes.
+  bool versioned = 8;
```

Comment:
Is it possible that a table contains both basic encoded and column-aware encoded rows?

Reply:
If we don't consider backward compatibility, then nope. Currently, we follow the `version` field of `TableCatalog` to decide the encoding.